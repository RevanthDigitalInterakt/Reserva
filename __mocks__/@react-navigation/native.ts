const navigate = jest.fn();

const useNavigation = () => ({
  navigate,
});

export {
  useNavigation,
};
