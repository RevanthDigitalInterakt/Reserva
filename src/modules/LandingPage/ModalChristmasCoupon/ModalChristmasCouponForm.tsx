import {
  Box,
  Typography,
  TextField,
  Button,
  Icon,
} from '@usereservaapp/reserva-ui';
import { useFormik } from 'formik';
import { Keyboard, Platform } from 'react-native';
import React, { useCallback, useEffect, useState } from 'react';
import * as yup from 'yup';
import ConfettiCannon from 'react-native-confetti-cannon';
import { theme } from '@usereservaapp/reserva-ui/src/themes';
import useMasterdataProvider, { IResponseSubmit } from '../../../hooks/useMasterdataProvider';
import { MasterDataSendDataDto } from '../../../types/dtos/masterdata-send-data.dto';
import { Colors } from '../../../Colors/Colors';
import useAsyncStorageProvider from '../../../hooks/useAsyncStorageProvider';
import { platformType } from '../../../utils/platformType';

const validationSchema = yup.object().shape({
  name: yup
    .string()
    .required('Informe um nome'),
  phone: yup
    .string()
    .required('Informe um celular')
    .matches(/^(\+?55\s?)?(\(\d{2}\)|\d{2})[\s-]?\d{5}-?\d{4}$/gi, {
      message: 'Informe um número de celular válido',
    }),
  email: yup
    .string()
    .required('Informe um e-mail')
    .matches(/^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$/g, {
      message: 'Informe um e-mail valido',
    }),
});

const inputStyle = {
  backgroundColor: Colors.INPUT_BACKGROUND,
  borderColor: Colors.INPUT_BORDER,
  color: Colors.INPUT_TEXT,
  borderWidth: 1,
  height: 40,
};

export interface IFormModel { email: string; name: string; phone: string; }

const formInitialState: IFormModel = { email: '', name: '', phone: '' };

export interface ModalChristmasCouponFormProps {
  title: string;
  subtitle: { content: string; fontFamily?: string }[];
  buttonTitle: string;
  fineline: string;
  onClose: () => void;
  orderId: string;
}

function ModalChristmasCouponForm({
  buttonTitle,
  title,
  subtitle,
  fineline,
  onClose,
  orderId,
}: ModalChristmasCouponFormProps) {
  const { setItem } = useAsyncStorageProvider();
  const { onSendData } = useMasterdataProvider();
  const [loading, setLoading] = useState<boolean>(false);
  const [feedback, setFeedback] = useState<IResponseSubmit>();

  const onSubmit = useCallback(async (values: IFormModel) => {
    if (loading) return;

    setLoading(true);

    Keyboard.dismiss();

    const dto = new MasterDataSendDataDto(
      values.email,
      values.name,
      values.phone,
      orderId,
    );

    setFeedback(await onSendData(dto));
    setLoading(false);
  }, [orderId, loading]);

  const formik = useFormik({
    initialValues: formInitialState,
    validationSchema,
    onSubmit,
  });

  useEffect(() => {
    if (feedback?.success) {
      setItem('@RNOrder:ChristmasCouponModalOrderId', '');
      formik.resetForm({ values: formInitialState });

      setTimeout(() => {
        onClose();
      }, 6000);
    }
  }, [feedback, onClose]);

  return (
    <Box
      mt={12}
      backgroundColor="white"
      minHeight={325}
      borderRadius={16}
      marginX="micro"
      style={{ elevation: Platform.OS === platformType.ANDROID ? 10 : 0 }}
      boxShadow={Platform.OS === platformType.ANDROID ? null : 'bottomBarShadow'}
    >
      <Box marginBottom={feedback?.success ? 30 : 42} marginTop={20} marginX={27}>
        <Box alignItems="center" flexDirection="row" justifyContent="space-between" marginBottom={5}>
          <Box flex={1}>
            <Typography
              fontSize={24}
              fontFamily="reservaSerifBlack"
              lineHeight={29}
              testID="com.usereserva:id/christmascouponform_title"
            >
              {title}
            </Typography>
          </Box>

          <Button
            onPress={onClose}
            variant="icone"
            icon={<Icon size={12} name="Close" />}
            testID="com.usereserva:id/christmascouponform_button_close"
          />
        </Box>

        <Box marginBottom={5}>
          <Typography
            fontSize={14}
            fontFamily="nunitoRegular"
            lineHeight={22}
            testID="com.usereserva:id/christmascouponform_subtitle"
          >
            {subtitle.map((item) => (
              <React.Fragment key={`text-${item.content}`}>
                <Typography fontFamily={(item.fontFamily || 'nunitoRegular') as keyof typeof theme.fonts}>
                  {item.content}
                </Typography>
              </React.Fragment>
            ))}
          </Typography>
        </Box>

        <Typography
          fontSize={12}
          fontFamily="nunitoBold"
          lineHeight={22}
          testID="com.usereserva:id/christmascouponform_fineline"
        >
          {fineline}
        </Typography>

        <Box marginTop={10}>
          <TextField
            style={inputStyle}
            value={formik.values.name}
            placeholder="NOME"
            height={40}
            onChangeText={(value) => formik.setFieldValue('name', value)}
            touched={formik.touched.name}
            error={formik.errors.name}
            accessibilityLabel="christmascouponform_input_name"
          />
        </Box>

        <Box marginTop={16}>
          <TextField
            style={inputStyle}
            value={formik.values.phone}
            maskType="cel-phone"
            placeholder="WHATSAPP"
            height={40}
            onChangeText={(value) => formik.setFieldValue('phone', value)}
            touched={formik.touched.phone}
            error={formik.errors.phone}
            accessibilityLabel="christmascouponform_input_phone"
          />
        </Box>

        <Box marginTop={16}>
          <TextField
            style={inputStyle}
            value={formik.values.email}
            touched={formik.touched.email}
            placeholder="E-MAIL"
            keyboardType="email-address"
            height={40}
            onChangeText={(value) => formik.setFieldValue('email', value)}
            error={formik.errors.email}
            accessibilityLabel="christmascouponform_input_email"
          />
        </Box>

        <Box>
          <Button
            inline
            height={40}
            marginTop={16}
            marginBottom={26}
            variant="primarioEstreito"
            loading={loading}
            title={buttonTitle || 'ASSINAR NEWSLETTER'}
            onPress={() => formik.submitForm()}
            testID="com.usereserva:id/christmascouponform_button_submit"
          />
        </Box>

        {!!feedback && (
          <Box mt={32}>
            <Typography
              fontFamily="nunitoSemiBold"
              fontSize={14}
              color={feedback.success ? 'verdeSucesso' : 'vermelhoAlerta'}
            >
              {feedback.title}
            </Typography>

            {!!feedback.message && (
              <Typography fontFamily="nunitoRegular" color="#3A3A3A">{feedback.message}</Typography>
            )}
          </Box>
        )}
      </Box>

      {!!feedback?.success && (
        <ConfettiCannon
          autoStart
          fadeOut
          count={150}
          origin={{ x: -10, y: 0 }}
        />
      )}
    </Box>
  );
}

export default ModalChristmasCouponForm;
