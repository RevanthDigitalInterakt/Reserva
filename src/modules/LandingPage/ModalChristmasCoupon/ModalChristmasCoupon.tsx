import React, {useCallback, useEffect, useMemo, useState} from 'react';
import Modal from 'react-native-modal';
import useMasterdataProvider from "../../../hooks/useMasterdataProvider";
import ModalChristmasCouponForm from "./ModalChristmasCouponForm";
import useAsyncStorageProvider from "../../../hooks/useAsyncStorageProvider";
import { IResponseCouponShowModal } from "../../../types/interfaces/IResponseCouponShowModal";

export interface IModalChristmasCoupon {
  isVisible: boolean;
  orderId: string;
  onClose: () => void;
}

function ModalChristmasCoupon({ isVisible, orderId, onClose }: IModalChristmasCoupon) {
  const { setItem } = useAsyncStorageProvider();
  const { onCheckChristmasModalVisibility } = useMasterdataProvider();

  const [modalInfo, setModalInfo] = useState<IResponseCouponShowModal>({
    showModal: false,
    title: '',
    subtitle: [],
    fineline: '',
    titleButton: '',
  });

  const modalIsVisible = useMemo(() => !!(
    modalInfo?.showModal && isVisible
  ), [modalInfo, isVisible])

  const onCloseModal = useCallback(() => {
    setItem('@RNOrder:ChristmasCouponModalOrderId', '');
    onClose();
  }, [onClose]);

  useEffect(() => {
    onCheckChristmasModalVisibility().then(setModalInfo)
  }, [])

  useEffect(() => {
    if (modalIsVisible && orderId) {
      setItem('@RNOrder:ChristmasCouponModalOrderId', orderId);
    }
  }, [modalIsVisible, orderId])

  return (
    <Modal
      avoidKeyboard
      onBackdropPress={onCloseModal}
      isVisible={modalIsVisible}
      testID="modalchristmas_container"
    >
      {!!modalInfo?.showModal && (
        <ModalChristmasCouponForm
          title={modalInfo.title}
          subtitle={modalInfo.subtitle}
          fineline={modalInfo.fineline}
          buttonTitle={modalInfo.titleButton}
          onClose={onCloseModal}
          orderId={orderId}
        />
      )}
    </Modal>
  );
}

export default ModalChristmasCoupon;
