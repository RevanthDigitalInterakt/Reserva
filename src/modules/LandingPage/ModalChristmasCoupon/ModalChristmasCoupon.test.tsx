import React from 'react';
import TestRenderer, { act } from 'react-test-renderer';
import { theme } from '@usereservaapp/reserva-ui';
import { ThemeProvider } from 'styled-components/native';
import ModalChristmasCoupon from './ModalChristmasCoupon';
import ModalChristmasCouponForm from './ModalChristmasCouponForm';

jest.mock('../../../hooks/useMasterdataProvider', () => () => ({
  onCheckChristmasModalVisibility: () => Promise.resolve({
    showModal: true,
    title: 'Titulo do modal',
    titleButton: 'Submit',
    subtitle: [],
    fineline: '*Confira as regras',
  }),
}));

jest.useFakeTimers();

const onClose = jest.fn();

const TestingComponent = (
  <ThemeProvider theme={theme}>
    <ModalChristmasCoupon isVisible onClose={onClose} orderId="123456" />
  </ThemeProvider>
);

describe('ModalChristmasCouponTest', () => {
  it('renders without error and match snapshot', async () => {
    await act(async () => {
      const renderer = await TestRenderer.create(TestingComponent);

      act(() => { jest.runAllTimers(); });
      await act(async () => {});

      const instance = renderer.root;

      act(() => { jest.runAllTimers(); });
      await act(async () => {});

      const $modal = instance.findByType(ModalChristmasCouponForm);
      const $title = instance.findByProps({ testID: 'christmascouponform_title' });

      expect($title.props.children).toBe('Titulo do modal');
      expect($modal).toBeTruthy();
      expect(renderer.toJSON()).toMatchSnapshot();
    });
  });

  it('hides the modal when onClose is called', async () => {
    await act(async () => {
      const renderer = await TestRenderer.create(TestingComponent);

      act(() => { jest.runAllTimers(); });
      await act(async () => {});

      const instance = renderer.root;

      act(() => { jest.runAllTimers(); });
      await act(async () => {});

      const modal = instance.findByProps({ testID: 'modalchristmas_container' });
      expect(modal.props.isVisible).toBeTruthy();

      const buttonClose = instance.findByProps({ testID: 'christmascouponform_button_close' });
      buttonClose.props.onPress();

      expect(onClose).toHaveBeenCalled();
    });
  });
});
