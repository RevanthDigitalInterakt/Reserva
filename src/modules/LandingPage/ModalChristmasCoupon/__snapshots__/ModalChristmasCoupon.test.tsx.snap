// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalChristmasCouponTest renders without error and match snapshot 1`] = `
<Modal
  animationType="none"
  deviceHeight={null}
  deviceWidth={null}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  panResponderThreshold={4}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  testID="com.usereserva:id/modalchristmas_container"
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0.7,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      Array [
        Array [
          Object {
            "margin": 37.5,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          },
          Object {
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
          Object {
            "margin": 0,
          },
        ],
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      deviceHeight={null}
      deviceWidth={null}
      forwardedRef={[Function]}
      hideModalContentWhileAnimating={false}
      onBackdropPress={[Function]}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      panResponderThreshold={4}
      pointerEvents="box-none"
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "margin": 37.5,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
    >
      <View
        backgroundColor="white"
        borderRadius={16}
        boxShadow="bottomBarShadow"
        marginX="micro"
        minHeight={325}
        mt={12}
        style={
          Array [
            Object {
              "backgroundColor": "#FFF",
              "borderRadius": 16,
              "marginLeft": 12,
              "marginRight": 12,
              "marginTop": 12,
              "minHeight": 325,
              "shadowColor": "rgba(0,0,0,0.1)",
              "shadowOffset": Object {
                "height": -3,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 2,
            },
            Object {
              "elevation": 0,
            },
          ]
        }
      >
        <View
          marginBottom={42}
          marginTop={20}
          marginX={27}
          style={
            Array [
              Object {
                "marginBottom": 42,
                "marginLeft": 27,
                "marginRight": 27,
                "marginTop": 20,
              },
            ]
          }
        >
          <View
            alignItems="center"
            flexDirection="row"
            justifyContent="space-between"
            marginBottom={5}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 5,
                },
              ]
            }
          >
            <View
              flex={1}
              style={
                Array [
                  Object {
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                  },
                ]
              }
            >
              <Text
                fontFamily="reservaSerifBlack"
                fontSize={24}
                lineHeight={29}
                style={
                  Array [
                    Object {
                      "fontFamily": "ReservaSerif-Black",
                      "fontSize": 24,
                      "lineHeight": 29,
                    },
                  ]
                }
                testID="com.usereserva:id/christmascouponform_title"
              >
                Titulo do modal
              </Text>
            </View>
            <View
              alignItems="center"
              style={
                Array [
                  Object {
                    "alignItems": "center",
                  },
                ]
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                    "opacity": 1,
                  }
                }
                testID="com.usereserva:id/christmascouponform_button_close"
              >
                <View
                  flexGrow={0}
                  justifyContent="flex-end"
                  style={
                    Array [
                      Object {
                        "flexGrow": 0,
                        "justifyContent": "flex-end",
                      },
                    ]
                  }
                >
                  <View
                    flexGrow={0}
                    justifyContent="center"
                    position="absolute"
                    right={-5}
                    style={
                      Array [
                        Object {
                          "flexGrow": 0,
                          "justifyContent": "center",
                          "position": "absolute",
                          "right": -5,
                          "top": -5,
                          "zIndex": 1,
                        },
                      ]
                    }
                    top={-5}
                    zIndex={1}
                  />
                  <RNSVGSvgView
                    bbHeight={12}
                    bbWidth={12}
                    color="preto"
                    focusable={false}
                    size={12}
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "borderWidth": 0,
                        },
                        Array [
                          Object {
                            "color": "#333333",
                            "height": 12,
                            "width": 12,
                          },
                        ],
                        Object {
                          "flex": 0,
                          "height": 12,
                          "width": 12,
                        },
                      ]
                    }
                  >
                    <RNSVGGroup>
                      <RNSVGSvgView
                        align="xMidYMid"
                        bbHeight="100%"
                        bbWidth="100%"
                        focusable={false}
                        height="100%"
                        meetOrSlice={0}
                        minX={0}
                        minY={0}
                        style={
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                              "borderWidth": 0,
                            },
                            Array [
                              Object {
                                "height": "100%",
                                "width": "100%",
                              },
                            ],
                            Object {
                              "flex": 0,
                              "height": "100%",
                              "width": "100%",
                            },
                          ]
                        }
                        vbHeight={8.24}
                        vbWidth={8.242}
                        width="100%"
                      >
                        <RNSVGGroup>
                          <RNSVGPath
                            d="M5.098 4.12l2.944-2.944A.69.69 0 007.066.2L4.122 3.144 1.178.2a.69.69 0 10-.976.976L3.146 4.12.202 7.064a.69.69 0 00.976.976l2.944-2.944L7.066 8.04a.69.69 0 00.976-.976z"
                            fill={4281545523}
                            propList={
                              Array [
                                "fill",
                              ]
                            }
                          />
                        </RNSVGGroup>
                      </RNSVGSvgView>
                    </RNSVGGroup>
                  </RNSVGSvgView>
                </View>
              </View>
            </View>
          </View>
          <View
            marginBottom={5}
            style={
              Array [
                Object {
                  "marginBottom": 5,
                },
              ]
            }
          >
            <Text
              fontFamily="nunitoRegular"
              fontSize={14}
              lineHeight={22}
              style={
                Array [
                  Object {
                    "fontFamily": "Nunito-Regular",
                    "fontSize": 14,
                    "lineHeight": 22,
                  },
                ]
              }
              testID="com.usereserva:id/christmascouponform_subtitle"
            />
          </View>
          <Text
            fontFamily="nunitoBold"
            fontSize={12}
            lineHeight={22}
            style={
              Array [
                Object {
                  "fontFamily": "Nunito-Bold",
                  "fontSize": 12,
                  "lineHeight": 22,
                },
              ]
            }
            testID="com.usereserva:id/christmascouponform_fineline"
          >
            *Confira as regras
          </Text>
          <View
            marginTop={10}
            style={
              Array [
                Object {
                  "marginTop": 10,
                },
              ]
            }
          >
            <View
              justifyContent="center"
              style={
                Array [
                  Object {
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <View
                backgroundColor="backgoundInput"
                borderColor="transparente"
                borderWidth="hairline"
                flexDirection="row"
                height={40}
                justifyContent="center"
                style={
                  Array [
                    Object {
                      "backgroundColor": "#F0F0F0",
                      "borderColor": "transparent",
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "height": 40,
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <View
                  flex={1}
                  style={
                    Array [
                      Object {
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                      },
                    ]
                  }
                >
                  <TextInput
                    accessibilityLabel="christmascouponform_input_name"
                    allowFontScaling={true}
                    flex={1}
                    onChangeText={[Function]}
                    paddingX="xxxs"
                    paddingY="nano"
                    placeholder="NOME"
                    placeholderTextColor="#8A8C8E"
                    rejectResponderTermination={true}
                    style={
                      Array [
                        Object {
                          "color": "#333333",
                          "flexBasis": 0,
                          "flexGrow": 1,
                          "flexShrink": 1,
                          "fontFamily": "Nunito-Regular",
                          "fontSize": 15,
                          "paddingBottom": 8,
                          "paddingLeft": 16,
                          "paddingRight": 16,
                          "paddingTop": 8,
                          "textAlign": "left",
                        },
                        Object {
                          "backgroundColor": "#ffffff",
                          "borderColor": "#BCBCBC",
                          "borderWidth": 1,
                          "color": "#3A3A3A",
                          "height": 40,
                        },
                      ]
                    }
                    textContentType="oneTimeCode"
                    underlineColorAndroid="transparent"
                    value=""
                    variant="paragraphSmall"
                  />
                </View>
                <View
                  justifyContent="center"
                  style={
                    Array [
                      Object {
                        "justifyContent": "center",
                      },
                    ]
                  }
                />
              </View>
            </View>
            <View
              style={
                Array [
                  Object {},
                ]
              }
            />
          </View>
          <View
            marginTop={16}
            style={
              Array [
                Object {
                  "marginTop": 16,
                },
              ]
            }
          >
            <View
              justifyContent="center"
              style={
                Array [
                  Object {
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <View
                backgroundColor="backgoundInput"
                borderColor="transparente"
                borderWidth="hairline"
                flexDirection="row"
                height={40}
                justifyContent="center"
                style={
                  Array [
                    Object {
                      "backgroundColor": "#F0F0F0",
                      "borderColor": "transparent",
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "height": 40,
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <View
                  flex={1}
                  style={
                    Array [
                      Object {
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                      },
                    ]
                  }
                >
                  <TextInput
                    accessibilityLabel="christmascouponform_input_phone"
                    allowFontScaling={true}
                    flex={1}
                    keyboardType="numeric"
                    onChangeText={[Function]}
                    paddingX="xxxs"
                    paddingY="nano"
                    placeholder="WHATSAPP"
                    placeholderTextColor="#8A8C8E"
                    rejectResponderTermination={true}
                    style={
                      Array [
                        Object {
                          "color": "#333333",
                          "flexBasis": 0,
                          "flexGrow": 1,
                          "flexShrink": 1,
                          "fontFamily": "Nunito-Regular",
                          "fontSize": 15,
                          "paddingBottom": 8,
                          "paddingLeft": 16,
                          "paddingRight": 16,
                          "paddingTop": 8,
                          "textAlign": "left",
                        },
                        Object {
                          "backgroundColor": "#ffffff",
                          "borderColor": "#BCBCBC",
                          "borderWidth": 1,
                          "color": "#3A3A3A",
                          "height": 40,
                        },
                      ]
                    }
                    textContentType="oneTimeCode"
                    type="cel-phone"
                    underlineColorAndroid="transparent"
                    value=""
                    variant="paragraphSmall"
                  />
                </View>
                <View
                  justifyContent="center"
                  style={
                    Array [
                      Object {
                        "justifyContent": "center",
                      },
                    ]
                  }
                />
              </View>
            </View>
            <View
              style={
                Array [
                  Object {},
                ]
              }
            />
          </View>
          <View
            marginTop={16}
            style={
              Array [
                Object {
                  "marginTop": 16,
                },
              ]
            }
          >
            <View
              justifyContent="center"
              style={
                Array [
                  Object {
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <View
                backgroundColor="backgoundInput"
                borderColor="transparente"
                borderWidth="hairline"
                flexDirection="row"
                height={40}
                justifyContent="center"
                style={
                  Array [
                    Object {
                      "backgroundColor": "#F0F0F0",
                      "borderColor": "transparent",
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "height": 40,
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <View
                  flex={1}
                  style={
                    Array [
                      Object {
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                      },
                    ]
                  }
                >
                  <TextInput
                    accessibilityLabel="christmascouponform_input_email"
                    allowFontScaling={true}
                    flex={1}
                    keyboardType="email-address"
                    onChangeText={[Function]}
                    paddingX="xxxs"
                    paddingY="nano"
                    placeholder="E-MAIL"
                    placeholderTextColor="#8A8C8E"
                    rejectResponderTermination={true}
                    style={
                      Array [
                        Object {
                          "color": "#333333",
                          "flexBasis": 0,
                          "flexGrow": 1,
                          "flexShrink": 1,
                          "fontFamily": "Nunito-Regular",
                          "fontSize": 15,
                          "paddingBottom": 8,
                          "paddingLeft": 16,
                          "paddingRight": 16,
                          "paddingTop": 8,
                          "textAlign": "left",
                        },
                        Object {
                          "backgroundColor": "#ffffff",
                          "borderColor": "#BCBCBC",
                          "borderWidth": 1,
                          "color": "#3A3A3A",
                          "height": 40,
                        },
                      ]
                    }
                    textContentType="oneTimeCode"
                    underlineColorAndroid="transparent"
                    value=""
                    variant="paragraphSmall"
                  />
                </View>
                <View
                  justifyContent="center"
                  style={
                    Array [
                      Object {
                        "justifyContent": "center",
                      },
                    ]
                  }
                />
              </View>
            </View>
            <View
              style={
                Array [
                  Object {},
                ]
              }
            />
          </View>
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <View
              alignItems="stretch"
              height={40}
              style={
                Array [
                  Object {
                    "alignItems": "stretch",
                    "height": 40,
                  },
                ]
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#333333",
                    "color": "#FFF",
                    "fontFamily": "Nunito-Bold",
                    "fontSize": 13,
                    "height": 40,
                    "justifyContent": "center",
                    "marginBottom": 26,
                    "marginTop": 16,
                    "minWidth": 130,
                    "opacity": 1,
                    "paddingBottom": 16,
                    "paddingLeft": 8,
                    "paddingRight": 8,
                    "paddingTop": 16,
                  }
                }
                testID="com.usereserva:id/christmascouponform_button_submit"
              >
                <View
                  alignItems="center"
                  flexDirection="row"
                  justifyContent="center"
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                    ]
                  }
                >
                  <View
                    height={40}
                    justifyContent="center"
                    style={
                      Array [
                        Object {
                          "height": 40,
                          "justifyContent": "center",
                        },
                      ]
                    }
                  >
                    <Text
                      color="white"
                      fontFamily="nunitoBold"
                      fontSize={13}
                      letterSpacing={1.6}
                      style={
                        Array [
                          Object {
                            "color": "#FFF",
                            "fontFamily": "Nunito-Bold",
                            "fontSize": 13,
                            "letterSpacing": 1.6,
                          },
                        ]
                      }
                    >
                      Submit
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
